WordPress/Classes/Services/Store.swift

```diff
@@ -22,53 +22,162 @@ class StoreKitTransactionObserver: NSObject, SKPaymentTransactionObserver {
+        
+        assert(transaction.payment.productIdentifier == pendingPayment.productID)
+        
+        guard let service = PlanService(siteID: pendingPayment.siteID, store: StoreKitStore()),
             let receiptURL = NSBundle.mainBundle().appStoreReceiptURL,
             let receipt = NSData(contentsOfURL: receiptURL)
             else {
                 assertionFailure()
                 return
         }
+        
         DDLogSwift.logInfo("[Store] Verifying purchase for transaction \(transaction)")
-        service.verifyPurchase(siteID, plan: plan, receipt: receipt, completion: { _ in
-            DDLogSwift.logInfo("[Store] Finishing transaction \(transaction)")
-            SKPaymentQueue.defaultQueue().finishTransaction(transaction)
+        service.verifyPurchase(pendingPayment.siteID, planID: pendingPayment.planID, receipt: receipt, completion: { [weak self] _ in
+            // TODO: Handle success / failure of verification attempt
+            self?.finishTransaction(transaction)
         })
     }
 
+    private func finishTransaction(transaction: SKPaymentTransaction) {
+        DDLogSwift.logInfo("[Store] Finishing transaction \(transaction)")
+
+        SKPaymentQueue.defaultQueue().finishTransaction(transaction)
+    
+        guard let productID = pendingPayment?.productID else { return }
+        
+        if transaction.payment.productIdentifier == productID {
+            pendingPayment = nil
+        }
+        
+        var userInfo: [String: AnyObject] = [StoreKitCoordinator.NotificationProductIdentifierKey: productID]
+        
+        if let error = transaction.error {
+            if error.code != SKErrorCode.PaymentCancelled.rawValue {
+                userInfo[NSUnderlyingErrorKey] = error
+            }
+            
+            postTransactionFailedNotification(userInfo)
+        } else {
+            postTransactionFinishedNotification(userInfo)
+        }
+    }
+    
+    private func postTransactionFailedNotification(userInfo: [NSObject: AnyObject]? = nil) {
+        NSNotificationCenter.defaultCenter().postNotificationName(StoreKitCoordinator.TransactionDidFailNotification,
+                                                                  object: nil,
+                                                                  userInfo: userInfo)
+    }
+    
+    private func postTransactionFinishedNotification(userInfo: [NSObject: AnyObject]? = nil) {
+        NSNotificationCenter.defaultCenter().postNotificationName(StoreKitCoordinator.TransactionDidFinishNotification,
+                                                                  object: nil,
+                                                                  userInfo: userInfo)
+    }
+    
+    /// - Returns: A PurchaseAvailability indicating whether the specified plan is currently
+    ///            available for purchase for the specified site, given the current active plan.
```

Comment:
I think every documented method should have a description. A `returns` callout should be added only if it adds information.

I'd swear I've seen a guideline like that in the past in the markup documentation, but can't find it right now.